### PR TITLE
Updates permissions and command listing for  'firebase ext' command.

### PR DESCRIPTION
**Pull request by joehan**
### Description
Corrects permissions and command listing for  'firebase ext' command. 

The command list portion of this command was broken by https://github.com/firebase/firebase-tools/pull/1784. This refactor leverages the new command.ts.

Additionally, I refactored this so that, when called without a project, it just prints out the commands instead of erroring.

### Scenarios Tested
On master, ```firebase ext``` gives:
Screen Shot 2019-11-19 at 9 51 19 AM

With the fix in this pr:
firebase ext -P project
Screen Shot 2019-11-19 at 3 15 57 PM

firebase ext
Screen Shot 2019-11-19 at 3 15 37 PM

### CODE REVIEW

**Comment by joehan** [comment]
Just chatted in person with Bryan about this. Looks like we broke this while typescriptifying command.ts, but we talked through a better way to fix this. For now, removing other reviewers as to not waste anyones time.

File: src/commands/ext.ts [hunk 1]
```diff
@@ -3,41 +3,40 @@ import * as clc from "cli-color";
...
-      "ext-configure",
-      "ext-info",
-      "ext-install",
-      "ext-list",
-      "ext-uninstall",
-      "ext-update",
+  .action(async (options: any) => {
+    // Print out help info for all extensions commands.
+    utils.logLabeledBullet(logPrefix, "list of extensions commands:")
+    const firebaseTools = require("../"); // eslint-disable-line @typescript-eslint/no-var-requires
```
Comment (joehan) [inline src/commands/ext.ts]: This is a circular dependency AND a linter violation, but after a bunch of searching, Bryan and I concluded that this was the best of a bunch of bad options.

File: src/commands/ext.ts [hunk 2]
```diff
@@ -3,41 +3,42 @@ import * as clc from "cli-color";
...
-      logger.info(`${clc.bold(cmd._cmd)} - ${cmd._description}`);
-      if (cmd._options.length > 0) {
-        logger.info("Option(s):");
-        _.forEach(cmd._options, (option) => {
-          logger.info("  ", option[0], " ", option[1]);
-        });
-      }
-      logger.info();
+    _.forEach(commandNames, (commandName) => {
+      const command: any = firebaseTools.getCommand(commandName);
```
Comment (bkendall) [inline src/commands/ext.ts]:
```typescript
import { CommanderStatic } from "commander";

// and then
const command: CommanderStatic = 
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,2 +1,3 @@
 * Improved error handling when exporting users.
-* Fixes an issue where repeated invoations cause an `EADDRINUSE` error (#1815)
\ No newline at end of file
+* Fixes firebase ext command and enables it to run without a project.
```
Comment (bkendall) [inline CHANGELOG.md]:
```suggestion
* Fixes Firebase `ext` command and enables it to run without a project.
```

**Pull request merged**